aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -0,0 +1,922 @@
+            );
+        };
+
+        // In case there's any dust left, send them all to the staker.
+        if (coin::value(&coins) > 0) {
+            coin::deposit(staker, coins);
+        } else {
+            coin::destroy_zero(coins);
+        }
+    }
+
+    // Assert that a staking_contract exists for the staker/operator pair.
+    fun assert_staking_contract_exists(staker: address, operator: address) acquires Store {
+        assert!(exists<Store>(staker), error::not_found(ENO_STAKING_CONTRACT_FOUND));
+        let staking_contracts = &mut borrow_global_mut<Store>(staker).staking_contracts;
+        assert!(
+            simple_map::contains_key(staking_contracts, &operator),
+            error::not_found(ESTAKING_CONTRACT_NOT_FOUND),
+        );
+    }
+
+    // Create a salt for generating the resource accounts that will be holding the StakePool for a given staking
+    // contract. This address should be deterministic for the same staker and operator.
+    fun create_seed(staker: address, operator: address): vector<u8> {
+        let seed = bcs::to_bytes(&staker);
+        vector::append(&mut seed, bcs::to_bytes(&operator));
+        // Include a salt to avoid conflicts with any other modules out there that might also generate
+        // deterministic resource accounts for the same staker + operator addresses.
+        vector::append(&mut seed, SALT);
+        seed
+    }
+
+    // Add a new distribution for `recipient` and `amount` to the staking contract's distributions list.
+    fun add_distribution(
+        operator: address,
+        staking_contract: &mut StakingContract,
+        recipient: address,
+        coins_amount: u64,
+        add_distribution_events: &mut EventHandle<AddDistributionEvent>,
+    ) {
+        // Update the total coins the pool is now worth first before adding the new distribution.
+        let distribution_pool = &mut staking_contract.distribution_pool;
+        let (_, _, _, total_distribution_amount) = stake::get_stake(staking_contract.pool_address);
+        pool_u64::update_total_coins(distribution_pool, total_distribution_amount);
+        pool_u64::buy_in(distribution_pool, recipient, coins_amount);
+
+        let pool_address = staking_contract.pool_address;
+        emit_event(
+            add_distribution_events,
+            AddDistributionEvent { operator, pool_address, amount: coins_amount }
+        );
+    }
+
+    // Calculate accumulated rewards and commissions since last update.
+    fun get_staking_contract_amounts_internal(staking_contract: &StakingContract): (u64, u64, u64) {
+        // Any outgoing flows of funds before the staker withdraws all staking_contracts can only come from operator
+        // withdrawing commissions.
+        // So to calculate rewards, we only care about current active + pending_active - last recorded principal.
+        let (active, _, pending_active, _) = stake::get_stake(staking_contract.pool_address);
```

Comment:
why doesn't this include `pending_inactive`?

Reply:
Because pending_inactive can only come from:
1. Outgoing commissions. This means commission has already been extracted.
2. Stake withdrawals from stakers. This also means commission has already been extracted as request_commission_internal is called in unlock_stake